src/Middleware/Spa/SpaProxy/src/SpaProxyLaunchManager.cs

```diff
@@ -0,0 +1,208 @@
+                if (OperatingSystem.IsWindows() && !Path.HasExtension(command))
+                {
+                    // On windows we transform npm/yarn to npm.cmd/yarn.cmd so that the command
+                    // can actually be found when we start the process. This is overridable if
+                    // necessary by explicitly setting up the extension on the command.
+                    command = $"{command}.cmd";
+                }
+
+                var info = new ProcessStartInfo(command, arguments)
+                {
+                    CreateNoWindow = false,
+                    UseShellExecute = true,
+                    WindowStyle = ProcessWindowStyle.Normal,
+                    WorkingDirectory = Path.Combine(AppContext.BaseDirectory, _options.WorkingDirectory)
+                };
+                _spaProcess = Process.Start(info);
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, $"Failed to launch the SPA development server '{_options.LaunchCommand}'.");
+            }
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
+                if (disposing)
+                {
+                    // Nothing to do here since ther are no managed resources
+                }
+
+                try
+                {
+                    if (_spaProcess != null)
+                    {
+                        // Review: Whether or not to do this at all. Turns out that if we try to kill the
+                        // npm.cmd/ps1 process that we start, even with this option we only stop this process
+                        // and the service keeps running.
+                        // Compared to performing Ctrl+C on the window or closing the window for the newly spawned
+                        // process which seems to do the right thing.
+                        // Process.CloseMainWindow seems to do the right thing in this situation and is doable since
+                        // we now start a proxy every time.
+                        // We can't guarantee that we stop/cleanup the proxy on every situation (for example if someone)
+                        // kills this process in a "rude" way, but this gets 95% there.
+                        // For cases where the proxy is left open and where there might not be a "visible" window the recomendation
+                        // is to kill the process manually. (We will not fail, we will simply notify the proxy is "already" up.
+                        if (!_spaProcess.CloseMainWindow())
+                        {
+                            _spaProcess.Kill(entireProcessTree: true);
+                            _spaProcess = null;
+                        }
```

Comment:
This have made me think that this code might not be 100% correct, since I'm not 100% sure I should mock with the `_spaProcess` instance when running in the context of the finalizer. That said, I don't think that matters since this object is managed by the DI container and will always be disposed, and the worst case scenario is an that an exception is thrown and caught within the finalizer, which I think it's fine.

I'm not going to change it since that would require me to retest all the templates across OSs again.